packages/selenium/test/RunPartial/TestBase.cs

```diff
@@ -0,0 +1,166 @@
+            );
+
+            dylangConfigPath = FixturePath("dylang-config.json");
+
+            driver = NewDriver();
+        }
+
+        [OneTimeTearDown]
+        public void TearDown()
+        {
+            driver.Close();
+            driver.Quit();
+        }
+
+        protected AxeBuilderOptions CustomSource(string axeSource)
+        {
+            return new AxeBuilderOptions
+            {
+                ScriptProvider = new StringAxeScriptProvider(axeSource)
+            };
+        }
+        protected void GoToResource(string url)
+        {
+            driver.Navigate().GoToUrl(ResourceUrl(url));
+        }
+
+        protected void GoToFixture(string url)
+        {
+            driver.Navigate().GoToUrl(FixtureUrl(url));
+        }
+
+        protected void GoToUrl(string url)
+        {
+            driver.Navigate().GoToUrl(url);
+        }
+
+        protected void AssertHasViolations(AxeResult results, params string[] rules)
+        {
+            var violations = results.Violations;
+            foreach (var violation in violations)
+            {
+                Console.WriteLine($">> {violation.Id}");
+            }
+
+            Assert.That(violations.Length, Is.EqualTo(rules.Length));
+
+            foreach (var rule in rules)
+            {
+                var found = false;
+                foreach (var vio in violations)
+                {
+                    if (vio.Id == rule)
+                    {
+                        found = true;
+                        break;
+                    }
+                }
+
+                if (!found)
+                {
+                    Console.Error.WriteLine($"Expected '{rule}' violation but could not find.");
+                    Assert.IsTrue(found);
+                }
+            }
+        }
+
+        private static void EnsureWebdriverPathInitialized(ref string driverPath, string dirEnvVar, string binaryName, IDriverConfig driverManagerConfig)
+        {
+            LazyInitializer.EnsureInitialized(ref driverPath, () =>
+            {
+                var dirFromEnv = Environment.GetEnvironmentVariable(dirEnvVar);
+                if (dirFromEnv != null)
+                {
+                    return $"{dirFromEnv}/${binaryName}";
+                }
+                else
+                {
+                    return new DriverManager().SetUpDriver(driverManagerConfig);
+                }
+            });
+        }
+        public static IWebDriver NewDriver()
+        {
+            EnsureWebdriverPathInitialized(ref ChromeDriverPath, "CHROMEWEBDRIVER", "chromedriver", new ChromeConfig());
+            ChromeDriverService chromeService = ChromeDriverService.CreateDefaultService(Path.GetDirectoryName(ChromeDriverPath));
+            chromeService.SuppressInitialDiagnosticInformation = true;
+            var options = new ChromeOptions();
+            if (Environment.GetEnvironmentVariable("CI") != null)

```

Comment:
This should be consistent with the existing logic in `IntegrationTests.cs` about which browser settings to use/how to detect whether to do headless mode/etc. Would maybe be best to separate out that part of `IntegrationTests.cs` into a `/test/TestBase.cs`, rename this class as `/test/RunPartial/RunPartialTestBase.cs`, and derive this one from the root `TestBase`.